core/src/main/scala/org/apache/spark/storage/ExternalBlockStore.scala

```diff
@@ -62,42 +62,50 @@ private[spark] class ExternalBlockStore(blockManager: BlockManager, executorId:
       values: Iterator[Any],
       level: StorageLevel,
       returnValues: Boolean): PutResult = {
-    logDebug(s"Attempting to write values for block $blockId")
-    val bytes = blockManager.dataSerialize(blockId, values)
-    putIntoExternalBlockStore(blockId, bytes, returnValues)
+    putIntoExternalBlockStore(blockId, Left(values), returnValues)
   }
 
   private def putIntoExternalBlockStore(
       blockId: BlockId,
-      bytes: ByteBuffer,
+      data: Either[Iterator[_], ByteBuffer],
```

Comment:
Please don't use Either. I think it's easier to just break this into two functions.